### PR TITLE
[FE: HM-4] - Angular Routing

### PR DESCRIPTION
**Pull request by ewgenij87snwork**
posts_404
posts_detail
posts_edit
preloader

### CODE REVIEW

File: src/app/modules/shared/components/header/header.component.scss [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  ::ng-deep .mat-progress-bar-buffer {
+    background: $white;
+  }
+
+  ::ng-deep.mat-progress-bar {
+    border-radius: 5px;
+  }
+
+}
```
Comment (MMaksymivUa) [inline src/app/modules/shared/components/header/header.component.scss:18]: you could create separate spinner component(even if you are using material preloading component) for customization and reusability in other's components

File: src/app/services/preloader/preloader.service.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  }
+
+  public hide(): void {
+    this.visible = false;
+  }
+
+  public show(): void {
+    this.visible = true;
+  }
+}
```
Comment (MMaksymivUa) [inline src/app/services/preloader/preloader.service.ts:22]:
```
    private spinnerSubject = new BehaviorSubject<boolean>(false);

    get isSpinnerVisible$() {
        return this.spinnerSubject.asObservable().pipe(distinctUntilChanged());
    }
    constructor() {}

    show(): void {
        this.spinnerSubject.next(true);
    }

    hide(): void {
        this.spinnerSubject.next(false);
    }
```
as variant

**Review by MMaksymivUa — APPROVED** [review]
Excellent. Good job, like how it looks

**Comment by ewgenij87snwork** [comment]
> Excellent. Good job, like how it looks

T

**Comment by ewgenij87snwork** [comment]
Thank You!

**Pull request closed without merging**